### ISSUE TITLE
Provide an easy way to create stateless pipes

### ISSUE THREAD

**Issue by vsavkin** [angular/angular#2902]
The pipes API is very flexible, which can be overwhelming when adding simple stateless pipes. We need to make it simpler.

Proposal:

```
class MyPipe extends SimplePipe {
  supports(obj) { return true; }
  transform(a, args) {
    
  }
}
```

**Comment by jeffbcross** [comment]
Removed from milestone and unassigned because @vsavkin suggested we all talk about this problem with @mhevery later.

**Comment by mhevery** [comment]
Closed by: 839edaa15b77a0b1e448f7a88c78f109a6b831fe
